broker-daemon/utils/migrate-store.js

```diff
@@ -7,20 +7,47 @@
  * @param  {number}   batchSize         - Number of operations to batch before executing on the target database
  * @returns {Promise<void>}
  */
-async function migrateStore (sourceStore, targetStore, createDbOperation, batchSize = 20) {
+async function migrateStore (sourceStore, targetStore, createDbOperation, batchSize = 2000) {
   return new Promise((resolve, reject) => {
     const stream = sourceStore.createReadStream()
+
+    // `batchInProgress` allows us to batch save records in sequence and lock the operation
+    // to make sure we are not inserting records out of the order they are received.
+    //
+    // We set this value to automatically resolve on the first use. This value is then
+    // subsequently set when a batch is processed
+    let batchInProgress = async () => {}
```

Comment:
you don't have to define this as a function, `await undefined` will immediately resolve.

Reply:
@treygriffith you have to call the function, so you'll receive `batchInProgress is not a function` wont you?